ygen/gogen.go

```diff
@@ -1123,7 +1127,14 @@ func findMapPaths(parent *yangStruct, field *yang.Entry, compressOCPaths bool) (
 			// is prefixed with a /.
 			// Absolute paths are used for top-level entities such rendering functions
 			// can be agnostic to whether the fakeroot was created.
-			childPath = append(childPath, []string{"", parentPath[2]}...)
+			// A special case exists where there are lists at the root, since in this case
+			// we cannot simply have the parent path prepended to the current element's
+			// name, and therefore we do not use absolute paths.
+			if parent.entry != nil && parent.entry.IsList() {
```

Comment:
Isn't this something that's more specific to json output? If so, would it be better to put special handling for lists there?

Reply:
I think it's more general -- although this comment in written around this case. For `Validate` - when we have compressed schemas, then we have `path:"foo/bar"` (where `bar` is the list) in the fakeroot - this allows you to find the `bar` schema. Prior to this change we have `path:""` in the fakeroot - such that `Validate` can't find an empty schema for the list there. Adding this here means that `Validate` can use `path` to find the list in the compressed and uncompressed cases.